core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -600,8 +600,14 @@ private InputStream getBlockInStream(long blockId) throws IOException {
         throw e;
       }
       long blockStart = BlockId.getSequenceNumber(blockId) * mBlockSize;
-      return createUnderStoreBlockInStream(blockStart, getBlockSize(blockStart),
-          mStatus.getUfsPath());
+      try {
+        return createUnderStoreBlockInStream(blockStart, getBlockSize(blockStart),
+            mStatus.getUfsPath());
+      } catch (IOException e2) {
+        LOG.debug("Failed to read from UFS after failing to read from Alluxio: {}", e2);
```

Comment:
I think this makes sense at the debug level since it's useful to know exactly where the exception came from, and the user explicitly set debug-level logging. If you're worried that it could be too verbose I don't mind switching to just the message. What do you think?